src/System.Net.WebSockets.WebSocketProtocol/tests/WebSocketProtocolTests.cs

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.WebSockets.Tests
+{
+    public sealed class WebSocketProtocolTests
```

Comment:
How are we testing the WebSocket returned from WebSocketProtocol.CreateFromStream?  The only test I see below that exercises it is WebSocketProtocol_CreateFromConnectedStream_Succeeds, which tests very little of it, e.g. it doesn't call ReceiveAsync at all.

Reply:
I had mentioned this in the issue post, re-commenting here, 

I was not able to get a successful test with WebSocket.ReceiveAsync, it fails in ReceiveAsync, because the response from the server is not in the right format. So the code fails with ProtocolError in ReceiveAsync. The behavior is same with and without this refactoring. Hence didn't include ReceiveAsync in the WebSocketProtocol_CreateFromConnectedStream_Succeeds outerloop test.

I can include this as a skipped test, to unblock this PR, and file a follow up issue to investigate the failure.